docs/design/dbaas-catalog-design.md

```diff
@@ -0,0 +1,785 @@
+-->
+
+TBD
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+[OLM Catalogs](https://olm.operatorframework.io/docs/concepts/crds/catalogsource/) is the way to distribute set of operator that can be installed onto a cluster via OLM.
+
+This Catalog will be based on  [Veneers](https://olm.operatorframework.io/docs/reference/veneers/), specifically on `semver veneer` as an approach to simplifying interacting with FBCs (File-Based Catalogs).
+
+Please see [Semver Veneer specification](https://olm.operatorframework.io/docs/reference/veneers/#semver-veneer).
+
+Catalog should have number of different testing layers to test:
+  - sanity of individual operator veneer/catalog
+  - integration test of new catalog with updated operator
+  - interoperability of the new operator version with other operators that could be affected
+  - e2e test for the platform configurations
+
+This repo provides only Catalog, OLM [operator bundles](https://olm.operatorframework.io/docs/tasks/creating-operator-bundle/) and their images are provided and maintained by individual operators.
+
+There could be number of bundle sources for this DBaaS Platform Catalog. Operator owners could:
+  1. create their own bundles and images
+  2. push operator's manifest and metadata to the [Kubernetes Community operators](https://github.com/k8s-operatorhub/community-operators) repo
+
+When in a first case Operator owners are responsible to create CI/CD pipelines to build and publish bundle images, in a second case Kubernetes Community does that.
+
+```mermaid
+C4Context
+    title Operator bundle Workflow for DBaaS Platform Catalog
+
+    Person(OperatorOwnerA, "Operator A")
+    Container(ImageRegA, "Operator A image Registry")
+
+    System_Boundary(c1, "DBaaS Platform Catalog") {
+        Container(DBaaSPlatformCatalog, "DBaaS Platform Catalog Image")
+    }
+
+    Container(RegOperatorHubIO, "Operator B in OperatorHub.io Registry")
+    Person(OperatorHubIO, "Operator B")
+
+    Rel_D(OperatorOwnerA, ImageRegA, "Creates bundle", "image")
+    Rel_D(ImageRegA, DBaaSPlatformCatalog, "Updates image in the catalog")
+
+    Rel_U(RegOperatorHubIO, DBaaSPlatformCatalog, "Updates image in the catalog", "image")
+    Rel_U(OperatorHubIO, RegOperatorHubIO, "Creates bundle", "image")
+```
+### Adding/Updating operator bundle
+
+To add new operator - create `veneer`:
+
+```sh
+mkdir catalog/percona-xtradb-cluster-operator ; cd catalog/percona-xtradb-cluster-operator
+cat << EOF >> veneer.yaml
+Schema: olm.semver
+GenerateMajorChannels: true
+GenerateMinorChannels: false
+Stable:
+  Bundles:
+  - Image: quay.io/operatorhubio/percona-xtradb-cluster-operator:v1.10.0
```

Comment:
I assume that for any operator that we develop there should be the release process that we have right now. An operator's team should create a new semver release tag for a new operator, bumping the version that was released, right? And in that case, we can use Fast (dev), Canditate (on release cut), and stable (Release) versions.

During the active development phase, operator image will be updated automatically using pipelines? Just to run tests against the newest version of an operator. Did I understand correctly the process of developing/testing operators?